src/main/java/org/openstreetmap/atlas/geography/atlas/dynamic/DynamicAtlas.java

```diff
@@ -138,6 +138,33 @@ public Rectangle bounds()
         return this.current.bounds();
     }
 
+    public void buildUnderlyingMultiAtlas()
+    {
+        logger.debug("{}: Loading MultiAtlas with {}", this.getName(),
+                nonNullShards().stream().map(Shard::getName).collect(Collectors.toList()));
+        this.policy.getShardSetChecker().accept(nonNullShards());
+        final List<Atlas> nonNullAtlasShards = getNonNullAtlasShards();
+        if (!nonNullAtlasShards.isEmpty())
+        {
+            if (nonNullAtlasShards.size() == 1)
+            {
+                this.current = nonNullAtlasShards.get(0);
+            }
+            else
+            {
+                this.current = new MultiAtlas(nonNullAtlasShards);
```

Comment:
instead of always creating a new multi atlas with all shards, is there a way we can create a multi atlas with the current DynamicAtlas and the new shards?  Would that help in performance?

Reply:
Every MultiAtlas creates some overhead and does some indexing of all the features in the sub-atlas it contains. So as this would not improve performance that much, it would increase memory consumption.